### PR TITLE
Add/remove message flag functions, event listener functions, store seqno with messages

### PR DESCRIPTION
**Pull request by redpandatronicsuk**
Hello,

I added wrapper functions to add and delete flags from messages, similar to the addMessageLabel function.

I also added event listener functions for receiving new mails, message updates (such as flag changes) and external message delete events. The functions are added to the config object, not sure if that is the best way to do it.

Also added to store the seqno along with messages, so we can identify messages when we receive the seqno from external delete events.

Cheers,
Dominik

### CODE REVIEW

File: lib/imapSimple.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ ImapSimple.prototype.openBox = function (boxName, callback) {
 ImapSimple.prototype.search = function (searchCriteria, fetchOptions, callback) {
     var self = this;
 
-    if (!callback && typeof(fetchOptions) === 'function') {
+    if (!callback && typeof (fetchOptions) === 'function') {
```
Comment (chadxz) [inline lib/imapSimple.js]: i agree with this change, but the parens should be removed because they're redundant

File: lib/imapSimple.js [hunk 2]
```diff
@@ -384,6 +443,19 @@ function connect(options, callback) {
...
+            imap.on('mail', options.onmail);
+        }
+
+        if (options.hasOwnProperty('onexpunge')) {
+            imap.on('expunge', options.onexpunge);
+        }
+
+        if (options.hasOwnProperty('onupdate')) {
+            imap.on('update', options.onupdate);
+        }
```
Comment (chadxz) [inline lib/imapSimple.js:457]: Do you mind adding documentation to describe when these events would be useful and the options to add listeners for them in the README?

File: lib/imapSimple.js [hunk 3]
```diff
@@ -319,6 +320,64 @@ ImapSimple.prototype.addMessageLabel = function (source, labels, callback) {
...
+        self.imap.delFlags(uid, flags, function (err) {
+            if (err) {
+                reject(err);
+                return;
+            }
+
+            resolve();
+        });
+    });
+};
```
Comment (chadxz) [inline lib/imapSimple.js:378]: can you please document these two methods in the README?

**Review by chadxz — CHANGES_REQUESTED** [review]
Changes look good, just a few comments.

File: lib/imapSimple.js (hunk 1 above)
Comment (redpandatronicsuk) [reply lib/imapSimple.js]: Didn't even know I made this change, must have been the VS Code auto-formatter

File: lib/imapSimple.js (hunk 3 above)
Comment (redpandatronicsuk) [reply lib/imapSimple.js:378]: Added something to the readme. Feel free to change the text if you don't like or can improve it

File: lib/imapSimple.js (hunk 2 above)
Comment (redpandatronicsuk) [reply lib/imapSimple.js:457]: Added something to the readme. Feel free to change the text if you don't like or can improve it

**Comment by redpandatronicsuk** [comment]
I made the changes. Not sure if they will ripple through or if I need to make another pull request. I'll make another one just in case.

**Comment by redpandatronicsuk** [comment]
The button now only says View Pull Request, instead of Create Pull Request, so I suppose the changes do go through

**Review by chadxz — APPROVED** [review]

**Comment by chadxz** [comment]
Thanks Dominik, published as imap-simple@3.1.0

**Comment by redpandatronicsuk** [comment]
Thanks for accepting the PR Chad!

**Pull request merged**